web_generator/lib/src/dart_main.dart

```diff
@@ -64,22 +68,51 @@ Future<void> generateJSInteropBindings({
 }
 
 Future<void> generateIDLBindings({
-  required String outputDirectory,
+  Iterable<String>? input,
+  required String output,
   required bool generateAll,
   required Version languageVersion,
 }) async {
-  const librarySubDir = 'dom';
-
-  ensureDirectoryExists('$outputDirectory/$librarySubDir');
-
-  final bindings = await generateBindings(packageRoot, librarySubDir,
-      generateAll: generateAll);
-  for (var entry in bindings.entries) {
-    final libraryPath = entry.key;
-    final library = entry.value;
-
-    final contents = _emitLibrary(library, languageVersion).toJS;
-    fs.writeFileSync('$outputDirectory/$libraryPath'.toJS, contents);
+  if (input == null) {
+    // parse dom library as normal
+    const librarySubDir = 'dom';
+
+    ensureDirectoryExists('$output/$librarySubDir');
+
+    final bindings = await generateBindings(packageRoot, librarySubDir,
+        generateAll: generateAll);
+
+    for (var entry in bindings.entries) {
+      final libraryPath = entry.key;
+      final library = entry.value;
+
+      final contents = _emitLibrary(library, languageVersion).toJS;
+      fs.writeFileSync('$output/$libraryPath'.toJS, contents);
+    }
+  } else {
+    final single = input.length == 1;
+    // parse individual files
+    ensureDirectoryExists(single ? p.dirname(output) : output);
+
+    final bindings = await generateBindingsForFiles({
+      for (final file in input)
+        file: fs
+            .readFileSync(file.toJS, JSReadFileOptions(encoding: 'utf-8'.toJS))
+            .dartify() as String
+    }, output, single);
+
+    for (var entry in bindings.entries.where((e) => e.key != 'dom.dart')) {
+      final libraryPath = entry.key;
+      final library = entry.value;
+
+      final contents = _emitLibrary(library, languageVersion).toJS;
+      // print('FOR $libraryPath \n${'='*20}\n $contents');
+      if (single) {
```

Comment:
I think we should always generate to a specified output directory if input files are specified. Assuming the user wants the output in a `lib/` folder but only in the non-single case is a bit awkward.